sphinxcontrib/needs/directives/needpie.py

```diff
@@ -156,10 +154,16 @@ def process_needpie(app, doctree, fromdocname):
                 if filter_func:
                     filter_func(**context)
                 sizes = context["results"]
+                # check items in sizes
+                for item in sizes:
+                    if not isinstance(item, int) and not isinstance(item, float):
+                        logger.error(f"{item} in sizes {sizes} is not a integer or float number.")
             except Exception as e:
                 raise e
-        if current_needpie["filter_func"] and content:
+        elif current_needpie["filter_func"] and content:
             logger.error("filter_func and content can't be used at the same time for needpie.")
+        else:
+            logger.error("Both filter_func and content not used for needpie.")
```

Comment:
I like the old message more:
"filter_func and content can't be used at the same time for needpie.""

Reply:
The old message is still there. This one is for the situation, where no content and no filter_func used at all.

Reply:
Ahh okay, was too stupid to read the diff the right way. all good.